lib/doorkeeper/models/access_token_mixin.rb

```diff
@@ -129,14 +129,36 @@ def generate_refresh_token
     end
 
     def generate_token
+      set_timestamps
+
       generator = Doorkeeper.configuration.access_token_generator.constantize
-      self.token = generator.generate(resource_owner_id: resource_owner_id,
-                                      scopes: scopes, application: application,
-                                      expires_in: expires_in)
+      self.token = generator.generate(
+        resource_owner_id: resource_owner_id,
+        scopes: scopes,
+        application: application,
+        expires_in: expires_in,
+        created_at: created_at
+      )
     rescue NoMethodError
       raise Errors::UnableToGenerateToken, "#{generator} does not respond to `.generate`."
     rescue NameError
       raise Errors::TokenGeneratorNotFound, "#{generator} not found"
     end
+
+    def set_timestamps
+      if created_at.nil?
+        self.created_at = current_time
+      end
```

Comment:
This method can be rewritten as `self.created_at ||= current_time`. In that case, we don't need it as a method, and I'd be comfortable adding it verbatim to line 132 where we call `set_timestamps`. What do you think?